odoo/addons/base/models/res_partner.py

```diff
@@ -348,7 +348,14 @@ def _compute_avatar_128(self):
     def _compute_avatar(self, avatar_field, image_field):
         partners_with_internal_user = self.filtered(
             lambda partner: partner.user_ids - partner.user_ids.filtered('share') or partner.type == 'contact')
-        super(ResPartner, partners_with_internal_user)._compute_avatar(avatar_field, image_field)
+        partners_to_get_initial_avatar = partners_with_internal_user.filtered(lambda ptr: ptr.user_ids and not ptr.id)
```

Comment:
Maybe lessen the load by not computing for partners that already have a computed avatar.
```suggestion
        partners_to_get_initial_avatar = partners_with_internal_user.filtered(
            lambda ptr: ptr.user_ids and not ptr.id and not ptr[avatar_field]
        )
```